### ISSUE TITLE
Formatter crash

### ISSUE THREAD

**Issue by benjaminwood** [cogini/logger_formatter_json#1]
Thanks for you work on this package! I started using it on a project recently and noticed some formatter crashes. Here's an example:

```
2024-04-03T00:46:36.704852+00:00 info: FORMATTER CRASH: {string,[<<"CONNECTED TO ">>,<<"Phoenix.LiveView.Socket">>,<<" in ">>,[<<"21">>,<<"µs"/utf8>>],<<"\n  Transport: ">>,<<":websocket">>,<<"\n  Serializer: ">>,<<"Phoenix.Socket.V2.JSONSerializer">>,<<"\n  Parameters: ">>,<<"%{\"_csrf_token\" => \"dhAeWGEbCAsdD2x9ZRoMaQEXZQBaWTF3ATAiXBYGYvTPTLgZxOUi73XE\", \"_live_referer\" => \"undefined\", \"_mounts\" => \"0\", \"_track_static\" => %{\"0\" => \"http://localhost:4000/assets/app.css\", \"1\" => \"http://localhost:4000/assets/app.js\"}, \"vsn\" => \"2.0.0\"}">>]}
```

Additionally, I noticed the formatter sometimes crashes when an error is being logged. I don't have an example of this at the moment, but I'll follow up and add one next time I see it.

Any ideas about what may be wrong? Here's our configuration:

```elixir
config :logger, :default_handler,
  formatter:
    {:logger_formatter_json,
     %{
       template: [
         :msg,
         :time,
         :level,
         :run_id,
         :runner_id,
         :run_extid
       ],
       names: %{
         time: "timestamp",
         msg: "message"
       }
     }}
```

**Comment by maxdrift** [comment]
Hello @reachfh and @benjaminwood! I encountered the same issue and after some digging figured out that the offending character is any special character e.g. `\n` after `µs` and probably any other non-ASCII time unit.
An Elixir test that reproduces the issue is:

```elixir
test "String with new line after microsecond" do
  expected = ~s({"msg":"408\\u00B5s","level":"info"}\n)

  assert expected ==
           to_string(
             :logger_formatter_json.format(
               %{
                 level: :info,
                 msg: {:string, "408µs\n"},
                 meta: %{}
               },
               %{}
             )
           )
end
```
failing with:
```
1) test Unstructured log messages String with new line after microsecond (LoggerFormatterJsonTest)
    test/logger_formatter_json_test.exs:29
    ** (ErlangError) Erlang error: {:invalid_byte, "0xB5", <<60, 60, 34, 52, 48, 56, 181, 115, 92, 110, 34, 47, 117, 116, 102, 56, 62, 62>>}
    code: :logger_formatter_json.format(
    stacktrace:
      (thoas 0.4.0) /some-path/logger_formatter_json/mix_tests/deps/thoas/src/thoas_encode.erl:1716: :thoas_encode.escape_unicode_chunk/5
      (thoas 0.4.0) /some-path/logger_formatter_json/mix_tests/deps/thoas/src/thoas_encode.erl:106: :thoas_encode.encode_string/2
      (thoas 0.4.0) /some-path/logger_formatter_json/mix_tests/deps/thoas/src/thoas_encode.erl:1703: :thoas_encode.map_naive/2
      (logger_formatter_json 0.8.1) /some-path/logger_formatter_json/src/logger_formatter_json.erl:65: :logger_formatter_json.format/2
      test/logger_formatter_json_test.exs:34: (test)
```

while the equivalent Erlang test would be:

```erlang
?assertEqual(
  <<"{\"msg\":\"793\\u00B5s\",\"level\":\"info\"}\n">>,
  iolist_to_binary(
    logger_formatter_json:format(
      #{level => info, msg => {string, [<<"793µs\n"/utf8>>]}, meta => #{}},
      #{}
    )
  )
),
```
failing with
```
%%% logger_formatter_json_SUITE:
%%% logger_formatter_json_SUITE ==> unstructured: FAILED
%%% logger_formatter_json_SUITE ==> {{invalid_byte,<<"0xB5">>,<<"<<\"793µs\\n\"/utf8>>">>},
 [{thoas_encode,escape_unicode_chunk,5,
                [{file,"/some-path/logger_formatter_json/mix_tests/_build/default/lib/thoas/src/thoas_encode.erl"},
                 {line,1716}]},
  {thoas_encode,encode_string,2,
                [{file,"/some-path/logger_formatter_json/mix_tests/_build/default/lib/thoas/src/thoas_encode.erl"},
                 {line,106}]},
  {thoas_encode,map_naive,2,
                [{file,"/some-path/logger_formatter_json/mix_tests/_build/default/lib/thoas/src/thoas_encode.erl"},
                 {line,1703}]},
  {logger_formatter_json,format,2,
                         [{file,"/some-path/logger_formatter_json/src/logger_formatter_json.erl"},
                          {line,65}]},
  {logger_formatter_json_SUITE,unstructured,1,
                               [{file,"/some-path/logger_formatter_json/test/logger_formatter_json_SUITE.erl"},
                                {line,37}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1302}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,1234}]}]}
```

**Comment by benjaminwood** [comment]
Nice, thanks @maxdrift!

Curious what @reachfh thinks about the best way to fix this sorta thing.

**Comment by maxdrift** [comment]
@reachfh I'm not seeing any more formatter crashes using `main` commit `f2b4177d0ac8338a79aa8f35b2191847ddc32e02`. Thank you so much! 🙇 
Looking forward to the new release! 🚢

**Comment by reachfh** [comment]
Fixed in 0.8.2

**Comment by reachfh** [comment]
Fixed in release 0.8.2
